cli/data/get.go

```diff
@@ -135,21 +138,38 @@ func copy(session *session.Session, source S3Path, destination string) {
 	}
 }
 
-func copyToLocalFiles(s3Client *s3.S3, objects []*s3.Object, source S3Path, destination string) {
+func copyToLocalFiles(s3Client *s3.S3, objects []*s3.Object, source S3Path, destination string, files []string) {
 	var (
-		wg  = new(sync.WaitGroup)
-		sem = make(chan struct{}, s3ParallelGets)
+		wg           = new(sync.WaitGroup)
+		sem          = make(chan struct{}, s3ParallelGets)
+		downloadList = filterObjects(objects, files)
 	)
 
 	wg.Add(len(objects))
 
-	for _, key := range objects {
+	for _, key := range downloadList {
 		go process(s3Client, source, destination, *key.Key, sem, wg)
 	}
 
 	wg.Wait()
 }
 
+func filterObjects(objects []*s3.Object, files []string) []*s3.Object {
+	var downloadList []*s3.Object
+	if len(files) == 0 {
+		return objects
+	}
+	for _, key := range objects {
+		_, file := filepath.Split(*key.Key)
```

Comment:
Will this work to fetch files that are within directories?

Reply:
Nope, as there is no recursion to search within directories. I believe we could re-visit this point if there is any use case.